src/openforms/js/components/admin/form_design/useConfirm.js

```diff
@@ -0,0 +1,57 @@
+import {useState} from 'react';
+import {useIntl} from 'react-intl';
+
+import ActionButton from 'components/admin/forms/ActionButton';
+import {Modal} from 'components/admin/modals';
+
+const useConfirm = (message, title = '') => {
+  const [promise, setPromise] = useState(null);
+
+  const confirm = () =>
+    new Promise(resolve => {
+      setPromise({resolve});
+    });
+
+  const handleClose = () => {
+    setPromise(null);
+  };
+
+  const handleConfirm = () => {
+    promise?.resolve(true);
+    handleClose();
+  };
+
+  const handleCancel = () => {
+    promise?.resolve(false);
+    handleClose();
+  };
+
+  const ConfirmationModal = () => {
+    const intl = useIntl();
+    const confirmBtnText = intl.formatMessage({
+      description: 'Confirmation modal confirm button',
+      defaultMessage: 'Confirm',
+    });
+    const cancelBtnText = intl.formatMessage({
+      description: 'Confirmation modal cancel button',
+      defaultMessage: 'Cancel',
+    });
+    return (
+      <Modal
+        title={title}
+        isOpen={promise !== null}
+        contentModifiers={['confirmation']}
+        closeModal={handleCancel}
+      >
+        <p>{message}</p>
+        <div className="react-modal__actions">
+          <ActionButton text={confirmBtnText} className="default" onClick={handleConfirm} />
+          <ActionButton text={cancelBtnText} onClick={handleCancel} />
+        </div>
+      </Modal>
+    );
+  };
+  return [ConfirmationModal, confirm];
```

Comment:
I remember reading that dynamically defining React components is an anti-pattern, and have tried this in the past in other places too and the conclusion was that it changes all the time/re-renders because there's no stable identity.

Utlimately, the proper solution is to return props to use for the component, and not return the whole component as is.

I'm afraid this approach here suffers from those same issues, and it may have been one of the reasons you were ending up in infinite render loops :thinking:

Reply:
Ah okay, that sound very interesting... Yeah then this very well might have played a bigger part in the whole infinite render issue..
I'll look into the article, and change the hook